modules/base/src/algorithm/volume/volumevoronoi.cpp

```diff
@@ -48,20 +60,26 @@ std::shared_ptr<Volume> voronoiSegmentation(
 
     util::forEachVoxelParallel(volumeDimensions, [&](const size3_t& voxelPos) {
         const auto transformedVoxelPos = mat3(indexToModelMatrix) * voxelPos;
-        auto minDist = std::numeric_limits<double>::max();
 
-        for (size_t i = 0; i < seedPointsWithIndices.size(); i++) {
-            // Squared distance
-            auto dist = glm::distance2(seedPointsWithIndices[i].second, transformedVoxelPos);
+        if (weightedVoronoi && weights.has_value()) {
```

Comment:
This is maybe nitpicking, but it is a good thing to consider.
I would move this if statements outside of the forEachVoxelParallel since they will have the same result for all voxels. You generally want to have as little code as possible inside of the "hot loop" i.e. the forEachVoxelParallel in this case.

Reply:
Good suggestion! I have fixed it now